src/dpp/commandhandler.cpp

```diff
@@ -421,20 +421,65 @@ void commandhandler::reply(const dpp::message &m, command_source source, command
 	}
 }
 
-void commandhandler::thinking(command_source source, command_completion_event_t callback)
+void commandhandler::thinking(command_source source, bool ephemeral, command_completion_event_t callback)
 {
 	dpp::message msg(this->owner);
 	msg.content = "*";
 	msg.guild_id = source.guild_id;
 	msg.channel_id = source.channel_id;
+	if (ephemeral) {
+		msg.set_flags(dpp::m_ephemeral);
+	}
 	if (!source.command_token.empty() && source.command_id) {
 		owner->interaction_response_create(source.command_id, source.command_token, dpp::interaction_response(ir_deferred_channel_message_with_source, msg), callback);
 	}
 }
 
-void commandhandler::thonk(command_source source, command_completion_event_t callback)
+void commandhandler::thonk(command_source source, bool ephemeral, command_completion_event_t callback)
 {
-	thinking(source, callback);
+	thinking(source, ephemeral, callback);
+}
+
+void commandhandler::edit_response(const message &m, command_source source, command_completion_event_t callback) {
```

Comment:
these all need a check like in thinking, e.g.:

```cpp
if (!source.command_token.empty() && source.command_id) {
```

they cant do anything if its a message command.